src/foreign_key.c

```diff
@@ -555,9 +566,13 @@ get_fk_index(Relation rel, int nkeys, AttrNumber *confkeys)
 			continue;
 		}
 
-		for (int i = 0; i < nkeys; i++)
+		for (int i = 0; i < nkeys && match; i++)
 		{
-			if (indexrel->rd_index->indkey.values[i] != confkeys[i])
+			/*
+			 * Since ordering of the constraint definition and index definition can differ,
+			 * we need to check that all the columns in the constraint are present in the index
+			 */
+			if (!array_has_member(confkeys, nkeys, indexrel->rd_index->indkey.values[i]))
```

Comment:
This check has quadratic complexity. Even though this is usually not a problem since the number of attributes is small it might be good to consider a check that is not quadratic.

Reply:
This check is not in a critical path and nkeys is capped at 32.